src/pages/POSSales.tsx

```
@@ -149,44 +154,81 @@ export default function POSSales() {
       bulkSelection.selectRange(dateFilteredSales, lastSelectedId, id);
     } else if (modifiers.isToggle) {
       bulkSelection.toggleItem(id);
     } else {
       bulkSelection.toggleItem(id);
     }
-    
+
     // Always update lastSelectedId to support subsequent range selections
     setLastSelectedId(id);
-  };
+  }, [lastSelectedId, bulkSelection, dateFilteredSales]);
```

Biome (2.3.13)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)